dali/server/daldap.cpp

```diff
@@ -144,24 +146,50 @@ class CDaliLdapConnection: implements IDaliLdapConnection, public CInterface
         Owned<ISecUser> user = ldapsecurity->createUser(username);
         user->credentials().setPassword(password);
 
-        //Check user's digital signature, if present
         bool authenticated = false;
-        if (!isEmptyString(udesc->querySignature()))
+
+        //Check that the digital signature provided by the caller (signature of
+        //caller's "username;timeStamp") matches what we expect it to be
+        if (!isEmptyString(reqSignature))
         {
             if (nullptr == pDSM)
                 pDSM = createDigitalSignatureManagerInstanceFromEnv();
             if (pDSM && pDSM->isDigiVerifierConfigured())
```

Comment:
Is it possible for client to have sent 'reqSignature' but for it to be unconfigured here?
(probably is - just checking)

Reply:
Yes it is possible, and unless they also send an LDAP username/pwd the call will fail. You might recall at the breakout meeting in Buckhead last year we discussed this as well as a signature enabled engine sending a request to a legacy Dali, and Richard seemed to agree that a failure was okay

Reply:
ok